cmd/root.go

```diff
@@ -167,14 +143,20 @@ var rootCmd = &cobra.Command{
 
 				if strings.ToLower(response) != "y\n" {
 					color.Green("Generated anchored Dockerfile\n")
-					fmt.Println(builder.String())
+					nodes.Write(os.Stdout)
 					return fmt.Errorf("exiting without writing file")
 				}
 			}
-			err = os.WriteFile(outputName, []byte(builder.String()), 0600)
+
+			f, err := os.Create(filepath.Clean(outputName))
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to create output file: %w", err)
 			}
+			defer f.Close()
+
+			w := bufio.NewWriter(f)
+			nodes.Write(w)
```

Comment:
This method has an `err` return I believe

Reply:
`nodes.Write(w)` ?

Reply:
I mean the `Write` function `io.Writer` with Write has a missing err return — we should bubble that up

Reply:
Ahh I see, fixed!